pyanaconda/modules/payloads/payload/rpm_ostree/flatpak_installation.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core.i18n import _
+from pyanaconda.modules.common.task import Task
+from pyanaconda.payload.errors import PayloadInstallError, FlatpakInstallError
+from pyanaconda.payload.flatpak import FlatpakPayload
+
+
+class InstallFlatpaksTask(Task):
+    """Task to install flatpaks from the local source on SilverBlue."""
+
+    def __init__(self, sysroot):
```

Comment:
The existing instance of `FlatpakPayload` is initialized in the `RPMOSTreePayloadWithFlatpaks` class, so you cannot create a new one for the task. However, it can be one of the arguments of the installation task.

Reply:
I don't think it matters in this case, same as for the OSTree stuff. As I see it:

The instance is persistent mainly to provide the install size on demand, but that could be easily extracted once and cached. The inputs are static, so the result must be too. In fact that would be a good refactoring step before further changes.

The remaining productive usage of the instance is preceded by `cleanup` which amounts to rmtree-ing whatever happened earlier and overwriting all private fields. A new instance will work just as well.

Reply:
I have verified the the caching idea and used it. The manager class now looks like a refactoring candidate, as I have identified two moderately independent code paths that it semi-randomly collects under one name. But that probably also includes a conversation with @jkonecny12 about potential future flatpak usage.

Reply:
I see. Thanks for the explanation!